pandas/util/_validators.py

```diff
@@ -269,6 +270,27 @@ def validate_bool_kwarg(
     return value
 
 
+def validate_na_arg(value, name: str):
+    """
+    Validate na arguments.
+
+    Parameters
+    ----------
+    value : object
+        Value to validate.
+    name : str
+        Name of the argument, used to raise an informative error message.
+
+    Raises
+    ______
+    ValueError
+        When ``value`` is determined to be invalid.
+    """
+    if value is lib.no_default or isinstance(value, bool) or isna(value):
```

Comment:
Can we be more specific than isna? E.g. I imagine pd.NaT or Decimal(“NaN”) would be weird?

Reply:
Aside from this, LGTM

Reply:
This would be a small breaking change, but no objection. Slight preference to keeping this PR just being an enforcement of the deprecation rather than a behavior change, but would be happy to make this change here. I think this just leaves `None`, `np.nan`, and `pd.NA` to check for, is that right? If so, plan to make the error message:

> f"{name} must be None, pd.NA, np.nan, True, or False; got {value}")